golang/x/relay/types/msgs.go

```diff
@@ -179,10 +179,10 @@ func (msg MsgNewRequest) Type() string { return "new_request" }
 // ValidateBasic runs stateless validation
 func (msg MsgNewRequest) ValidateBasic() sdk.Error {
 	// TODO: validate output types
-	if len(msg.Spends) != 36 {
+	if len(msg.Spends) != 36  && len(msg.Spends) != 0 {
 		return ErrSpendsLength(DefaultCodespace)
 	}
-	if len(msg.Pays) > 50 {
+	if len(msg.Pays) > 50 && len(msg.Spends) != 0 {
```

Comment:
should read `if len(msg.Pays) > 50`.  No need for the second check at all

Reply:
😂